app/components/bali/hover_card/component.rb

```diff
@@ -18,16 +18,9 @@ def initialize(hover_url: nil, placement: 'auto', **options)
         @data = options.delete(:data) || {}
         @hover_url = options.dig(:data, :'hovercard-url-value') || hover_url
         @options = prepend_class_name(options, 'hover-card-component')
-      end
-
-      private
-
-      def hovercard_data
-        {
-          controller: "hovercard #{data.delete(:controller)}",
-          'hovercard-placement-value': placement,
-          'hovercard-url-value': hover_url
-        }.merge!(data).compact
+        @options = prepend_controller(@options, "hovercard #{data.delete(:controller)}")
```

Comment:
No es necesario, el método **prepend_controller** te respeta si ya traes un controlador

```suggestion
        @options = prepend_controller(@options, "hovercard")
```